diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index dfe9f04..14be931 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -2,6 +2,7 @@
 
 namespace App\Controller;
 
+use App\Entity\User;
 use App\Repository\StarshipRepository;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Bundle\SecurityBundle\Security;
@@ -21,7 +22,12 @@ class MainController extends AbstractController
         $ships->setMaxPerPage(5);
         $ships->setCurrentPage($request->query->get('page', 1));
 
-        $myShip = $security->getUser()?->getStarship();
+        $myShip = null;
+        $user = $security->getUser();
+
+        if ($user instanceof User) {
+            $myShip = $user->getStarship();
+        }
 
         return $this->render('main/homepage.html.twig', [
             'myShip' => $myShip,
