diff --git a/src/Entity/User.php b/src/Entity/User.php
index 0d9e208..0b08125 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -7,6 +7,7 @@ use Doctrine\ORM\Mapping as ORM;
 use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Validator\Constraints as Assert;
 
 #[ORM\Entity(repositoryClass: UserRepository::class)]
 #[ORM\UniqueConstraint(name: 'UNIQ_IDENTIFIER_EMAIL', fields: ['email'])]
@@ -18,6 +19,8 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     #[ORM\Column]
     private ?int $id = null;
 
+    #[Assert\NotBlank]
+    #[Assert\Email]
     #[ORM\Column(length: 180)]
     private ?string $email = null;
 
@@ -33,6 +36,7 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     #[ORM\Column]
     private ?string $password = null;
 
+    #[Assert\NotBlank]
     #[ORM\Column(length: 255)]
     private ?string $name = null;
 
diff --git a/templates/registration/register.html.twig b/templates/registration/register.html.twig
index ef2cfb5..06f2e5b 100644
--- a/templates/registration/register.html.twig
+++ b/templates/registration/register.html.twig
@@ -14,7 +14,7 @@
 
                 {{ form_errors(registrationForm) }}
 
-                {{ form_start(registrationForm, { attr: { class: 'space-y-6' } }) }}
+                {{ form_start(registrationForm, { attr: { novalidate: true, class: 'space-y-6' } }) }}
                     {{ form_row(registrationForm.email, { label_attr: { class: 'text-sm font-medium text-white/80' } }) }}
                     {{ form_row(registrationForm.name, { label_attr: { class: 'text-sm font-medium text-white/80' } }) }}
                     {{ form_row(registrationForm.plainPassword, {
