diff --git a/src/Controller/LazyController.php b/src/Controller/LazyController.php
new file mode 100644
index 0000000..752a649
--- /dev/null
+++ b/src/Controller/LazyController.php
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Controller;
+
+use App\Repository\StarshipPartRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+final class LazyController extends AbstractController
+{
+    #[Route('/lazy', name: 'app_lazy')]
+    public function index(StarshipPartRepository $repository): Response
+    {
+        $part = $repository->find(1);
+
+        dump($part);
+
+        return $this->render('lazy/index.html.twig', [
+            'controller_name' => 'LazyController',
+        ]);
+    }
+}
diff --git a/templates/lazy/index.html.twig b/templates/lazy/index.html.twig
new file mode 100644
index 0000000..cfe2786
--- /dev/null
+++ b/templates/lazy/index.html.twig
@@ -0,0 +1,20 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Hello LazyController!{% endblock %}
+
+{% block body %}
+<style>
+    .example-wrapper { margin: 1em auto; max-width: 800px; width: 95%; font: 18px/1.5 sans-serif; }
+    .example-wrapper code { background: #F5F5F5; padding: 2px 6px; }
+</style>
+
+<div class="example-wrapper">
+    <h1>Hello {{ controller_name }}! ✅</h1>
+
+    This friendly message is coming from:
+    <ul>
+        <li>Your controller at <code>/Users/valkyrie/www/sf8/src/Controller/LazyController.php</code></li>
+        <li>Your template at <code>/Users/valkyrie/www/sf8/templates/lazy/index.html.twig</code></li>
+    </ul>
+</div>
+{% endblock %}
