diff --git a/templates/base.html.twig b/templates/base.html.twig
index 619324b..bc375dd 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -66,6 +66,7 @@
                         </a>
                     </nav>
                 </header>
+                {{ dump(app.user) }}
                 {% block body %}{% endblock %}
             </div>
             <div class="p-5 bg-white/5 mt-3 text-center">
