diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 9ee8073..b3688af 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -26,7 +26,7 @@
                                         class="hover:text-slate-200"
                                         href="#"
                                     >
-                                        {{ ship.type }}
+                                        {{ ship.type|replace({'_': ' '})|title }}
                                         {{ ship.name }}
                                     </a>
                                 </h4>
