diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 1b04e55..286518a 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -24,7 +24,7 @@ class StarshipRepository extends ServiceEntityRepository
     {
         return $this->createQueryBuilder('s')
             ->where('s INSTANCE OF :class')
-            ->setParameter('class', Scout::class)
+            ->setParameter('class', $this->getEntityManager()->getClassMetadata(Scout::class))
             ->getQuery()
             ->execute()
         ;
