diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 3219599..6d510c2 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -13,7 +13,7 @@
 
             <div class="space-y-5">
                 {% for ship in ships %}
-                    {{ include('starship/teaser.html.twig', {ship: ship}) }}
+                    {{ include('starship/teaser/'~ship.type~'.html.twig', {ship: ship}) }}
                 {% endfor %}
             </div>
         </div>
diff --git a/templates/starship/teaser.html.twig b/templates/starship/teaser.html.twig
index 3fb8d30..1134db4 100644
--- a/templates/starship/teaser.html.twig
+++ b/templates/starship/teaser.html.twig
@@ -18,6 +18,7 @@
             <div>
                 Arrived at: {{ ship.arrivedAt|ago }}
             </div>
+            {% block extra %}{% endblock %}
         </div>
     </div>
     <div class="flex justify-center min-[1174px]:justify-start mt-2 min-[1174px]:mt-0 shrink-0">
diff --git a/templates/starship/teaser/freighter.html.twig b/templates/starship/teaser/freighter.html.twig
new file mode 100644
index 0000000..48ba522
--- /dev/null
+++ b/templates/starship/teaser/freighter.html.twig
@@ -0,0 +1,5 @@
+{% extends 'starship/teaser.html.twig' %}
+
+{% block extra %}
+    <div>Cargo Capacity: {{ ship.cargoCapacity }}</div>
+{% endblock %}
