diff --git a/src/Security/Voter/SuperAdminVoter.php b/src/Security/Voter/SuperAdminVoter.php
index 3fc062f..556f22a 100644
--- a/src/Security/Voter/SuperAdminVoter.php
+++ b/src/Security/Voter/SuperAdminVoter.php
@@ -3,14 +3,24 @@
 namespace App\Security\Voter;
 
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\AuthenticatedVoter;
 use Symfony\Component\Security\Core\Authorization\Voter\Vote;
 use Symfony\Component\Security\Core\Authorization\Voter\Voter;
 
 final class SuperAdminVoter extends Voter
 {
+    private const EXCLUSIONS = [
+        AuthenticatedVoter::IS_AUTHENTICATED,
+        AuthenticatedVoter::IS_AUTHENTICATED_FULLY,
+        AuthenticatedVoter::IS_AUTHENTICATED_REMEMBERED,
+        AuthenticatedVoter::IS_IMPERSONATOR,
+        AuthenticatedVoter::IS_REMEMBERED,
+        AuthenticatedVoter::PUBLIC_ACCESS,
+    ];
+
     protected function supports(string $attribute, mixed $subject): bool
     {
-        return true;
+        return !in_array($attribute, self::EXCLUSIONS, true);
     }
 
     protected function voteOnAttribute(string $attribute, mixed $subject, TokenInterface $token, ?Vote $vote = null): bool
