diff --git a/config/packages/security.yaml b/config/packages/security.yaml
index 51d04a9..96c4533 100644
--- a/config/packages/security.yaml
+++ b/config/packages/security.yaml
@@ -26,6 +26,7 @@ security:
                 login_path: app_login
                 check_path: app_login
                 enable_csrf: true
+                target_path_parameter: referrer
             logout:
                 path: app_logout
                 enable_csrf: true
diff --git a/templates/base.html.twig b/templates/base.html.twig
index 434034d..36b0186 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -7,6 +7,8 @@
         {% block stylesheets %}
         {% endblock %}
 
+        {% block metadata %}{% endblock %}
+
         {% block javascripts %}
             {% block importmap %}{{ importmap('app') }}{% endblock %}
         {% endblock %}
@@ -84,7 +86,7 @@
                             <a class="hover:text-amber-400 pt-2" href="{{ path('app_register') }}">
                                 Register
                             </a>
-                            <a class="hover:text-amber-400 pt-2" href="{{ path('app_login', { _target_path: app.request.pathInfo }) }}">
+                            <a class="hover:text-amber-400 pt-2" href="{{ path('app_login', { referrer: app.request.pathInfo }) }}">
                                 Login
                             </a>
                         {% endif %}
diff --git a/templates/security/login.html.twig b/templates/security/login.html.twig
index f6a2ed5..d0254b0 100644
--- a/templates/security/login.html.twig
+++ b/templates/security/login.html.twig
@@ -2,6 +2,10 @@
 
 {% block title %}Sign in{% endblock %}
 
+{% block metadata %}
+    <link rel="canonical" href="{{ url('app_login') }}">
+{% endblock %}
+
 {% block body %}
     <section class="px-6 py-10 sm:py-16">
         <div class="mx-auto max-w-md">
