diff --git a/config/packages/security.yaml b/config/packages/security.yaml
index 48e8577..ebd39f7 100644
--- a/config/packages/security.yaml
+++ b/config/packages/security.yaml
@@ -28,6 +28,7 @@ security:
                 enable_csrf: true
                 # where to redirect after logout
                 # target: app_any_route
+            remember_me: ~
 
             # Activate different ways to authenticate:
             # https://symfony.com/doc/current/security.html#the-firewall
diff --git a/templates/security/login.html.twig b/templates/security/login.html.twig
index 49dfa1e..bc70043 100644
--- a/templates/security/login.html.twig
+++ b/templates/security/login.html.twig
@@ -54,15 +54,10 @@
 
                     <input type="hidden" name="_csrf_token" data-controller="csrf-protection" value="{{ csrf_token('authenticate') }}">
 
-                    {#
-                        Uncomment this section and add a remember_me option below your firewall to activate remember me functionality.
-                        See https://symfony.com/doc/current/security/remember_me.html
-
-                        <div class="flex items-center gap-3 rounded-lg border border-white/10 bg-white/5 px-4 py-3 text-sm text-white/80">
-                            <input type="checkbox" name="_remember_me" id="_remember_me" class="rounded border-white/20 bg-transparent text-cyan-400 focus:ring-cyan-400">
-                            <label for="_remember_me">Remember me</label>
-                        </div>
-                    #}
+                    <div class="flex items-center gap-3 rounded-lg border border-white/10 bg-white/5 px-4 py-3 text-sm text-white/80">
+                        <input type="checkbox" checked name="_remember_me" id="_remember_me" class="rounded border-white/20 bg-transparent text-cyan-400 focus:ring-cyan-400">
+                        <label for="_remember_me">Remember me</label>
+                    </div>
 
                     <button class="w-full rounded-lg bg-green-700 px-5 py-3 font-semibold text-white transition hover:bg-green-800 cursor-pointer" type="submit">
                         Sign in
