diff --git a/src/Controller/StarshipController.php b/src/Controller/StarshipController.php
index 273a6f1..7dbbd70 100644
--- a/src/Controller/StarshipController.php
+++ b/src/Controller/StarshipController.php
@@ -2,6 +2,7 @@
 
 namespace App\Controller;
 
+use App\Entity\Scout;
 use App\Entity\Starship;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
@@ -10,7 +11,7 @@ use Symfony\Component\Routing\Attribute\Route;
 final class StarshipController extends AbstractController
 {
     #[Route('/starship/{id}', name: 'app_starship_show')]
-    public function index(Starship $ship): Response
+    public function index(Scout $ship): Response
     {
         dd($ship);
 
