diff --git a/src/Controller/StarshipController.php b/src/Controller/StarshipController.php
index 273a6f1..efde9ac 100644
--- a/src/Controller/StarshipController.php
+++ b/src/Controller/StarshipController.php
@@ -12,10 +12,8 @@ final class StarshipController extends AbstractController
     #[Route('/starship/{id}', name: 'app_starship_show')]
     public function index(Starship $ship): Response
     {
-        dd($ship);
-
-        return $this->render('starship/index.html.twig', [
-            'controller_name' => 'StarshipController',
+        return $this->render('starship/show.html.twig', [
+            'ship' => $ship,
         ]);
     }
 }
diff --git a/templates/starship/index.html.twig b/templates/starship/index.html.twig
deleted file mode 100644
index 9677a43..0000000
--- a/templates/starship/index.html.twig
+++ /dev/null
@@ -1,20 +0,0 @@
-{% extends 'base.html.twig' %}
-
-{% block title %}Hello StarshipController!{% endblock %}
-
-{% block body %}
-<style>
-    .example-wrapper { margin: 1em auto; max-width: 800px; width: 95%; font: 18px/1.5 sans-serif; }
-    .example-wrapper code { background: #F5F5F5; padding: 2px 6px; }
-</style>
-
-<div class="example-wrapper">
-    <h1>Hello {{ controller_name }}! ✅</h1>
-
-    This friendly message is coming from:
-    <ul>
-        <li>Your controller at <code>/Users/valkyrie/www/doctrine-inheritance/src/Controller/StarshipController.php</code></li>
-        <li>Your template at <code>/Users/valkyrie/www/doctrine-inheritance/templates/starship/index.html.twig</code></li>
-    </ul>
-</div>
-{% endblock %}
diff --git a/templates/starship/show.html.twig b/templates/starship/show.html.twig
new file mode 100644
index 0000000..d6d5caa
--- /dev/null
+++ b/templates/starship/show.html.twig
@@ -0,0 +1,41 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}{{ ship.name }}{% endblock %}
+
+{% block body %}
+    <div class="bg-[#16202A] rounded-2xl pl-5 py-5 pr-11 flex flex-col min-[1174px]:flex-row min-[1174px]:justify-between">
+        <div class="flex justify-center min-[1174px]:justify-start">
+            <img class="h-[83px] w-[84px]" src="{{ asset(ship.statusImageFilename) }}" alt="Status: {{ ship.statusString }}">
+            <div class="ml-5">
+                <div class="rounded-2xl py-1 px-3 flex justify-center w-32 items-center bg-amber-400/10">
+                    <div class="rounded-full h-2 w-2 bg-amber-400 blur-[1px] mr-2"></div>
+                    <p class="uppercase text-xs text-nowrap">{{ ship.statusString }}</p>
+                </div>
+                <h4 class="text-[22px] pt-1 font-semibold">
+                    <a
+                        class="hover:text-slate-200"
+                        href="{{ path('app_starship_show', {id: ship.id}) }}"
+                    >
+                        {{ ship.type|replace({'_': ' '})|title }}
+                        {{ ship.name }}
+                    </a>
+                </h4>
+                <div>
+                    Arrived at: {{ ship.arrivedAt|ago }}
+                </div>
+                {% block extra %}{% endblock %}
+            </div>
+        </div>
+        <div class="flex justify-center min-[1174px]:justify-start mt-2 min-[1174px]:mt-0 shrink-0">
+            <div class="border-r border-white/20 pr-8">
+                <p class="text-slate-400 text-xs text-right">Captain</p>
+                <p class="text-xl">{{ ship.captain }}</p>
+            </div>
+
+            <div class="pl-8 w-[100px]">
+                <p class="text-slate-400 text-xs">Class</p>
+                <p class="text-xl">{{ ship.class }}</p>
+            </div>
+        </div>
+    </div>
+{% endblock %}
