diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 3c538bc..ac03190 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -25,7 +25,7 @@ class StarshipRepository extends ServiceEntityRepository
     public function filterShips(): array
     {
         return $this->createQueryBuilder('s')
-            ->where('s.ship_type = :class')
+            ->where('s = :class')
             ->setParameter('class', $this->getEntityManager()->getClassMetadata(Freighter::class))
             ->getQuery()
             ->execute()
