diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index f96e8c2..410edd8 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -14,10 +14,10 @@ class MainController extends AbstractController
 {
     #[Route('/', name: 'app_homepage')]
     public function homepage(
-        MiningFreighterRepository $starshipRepository,
+        StarshipRepository $starshipRepository,
         StarshipPartRepository $starshipPartRepository,
     ): Response {
-        $ships = $starshipRepository->findAll();
+        $ships = $starshipRepository->filterShips();
         $myShip = $ships[array_rand($ships)];
 
         dump($starshipPartRepository->find(1));
diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 010c073..ac03190 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -25,10 +25,8 @@ class StarshipRepository extends ServiceEntityRepository
     public function filterShips(): array
     {
         return $this->createQueryBuilder('s')
-            ->where('s INSTANCE OF :class')
-            ->andWhere('s NOT INSTANCE OF :notclass')
+            ->where('s = :class')
             ->setParameter('class', $this->getEntityManager()->getClassMetadata(Freighter::class))
-            ->setParameter('notclass', $this->getEntityManager()->getClassMetadata(MiningFreighter::class))
             ->getQuery()
             ->execute()
         ;
