diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 61008ed..e77a7ee 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -118,7 +118,7 @@ abstract class Starship
         };
     }
 
-    final public function getType(): string
+    final public static function getType(): string
     {
         return array_flip(self::TYPE_MAP)[static::class];
     }
diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 9938cd2..682e0a5 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -26,7 +26,7 @@ class StarshipRepository extends ServiceEntityRepository
     {
         return $this->createQueryBuilder('s')
             ->where('TYPE(s) = :class')
-            ->setParameter('class', 'freighter')
+            ->setParameter('class', Freighter::getType())
             ->getQuery()
             ->execute()
         ;
