diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
new file mode 100644
index 0000000..631c159
--- /dev/null
+++ b/src/Controller/StarshipApiController.php
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+class StarshipApiController extends AbstractController
+{
+    #[Route('/api/starships')]
+    public function getCollection(): Response
+    {
+        $starships = [
+            [
+                'name' => 'USS LeafyCruiser (NCC-0001)',
+                'class' => 'Garden',
+                'captain' => 'Jean-Luc Pickles',
+            ],
+            [
+                'name' => 'USS Espresso (NCC-1234-C)',
+                'class' => 'Latte',
+                'captain' => 'James T. Quick!',
+            ],
+            [
+                'name' => 'USS Wanderlust (NCC-2024-W)',
+                'class' => 'Delta Tourist',
+                'captain' => 'Kathryn Journeyway',
+            ],
+        ];
+
+        return $this->json($starships);
+    }
+}
