diff --git a/templates/admin/starship-part/new.html.twig b/templates/admin/starship-part/new.html.twig
index 55a4839..025dcac 100644
--- a/templates/admin/starship-part/new.html.twig
+++ b/templates/admin/starship-part/new.html.twig
@@ -1,5 +1,7 @@
 {% extends 'base.html.twig' %}
 
+{% form_theme form 'tailwind_2_layout.html.twig' %}
+
 {% block title %}Admin : New Starship Part{% endblock %}
 
 {% block body %}
