diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index e2b080a..9e8bc85 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -19,7 +19,7 @@
                             <div class="ml-5">
                                 <div class="rounded-2xl py-1 px-3 flex justify-center w-32 items-center bg-amber-400/10">
                                     <div class="rounded-full h-2 w-2 bg-amber-400 blur-[1px] mr-2"></div>
-                                    <p class="uppercase text-xs text-nowrap">in progress</p>
+                                    <p class="uppercase text-xs text-nowrap">{{ ship.status }}</p>
                                 </div>
                                 <h4 class="text-[22px] pt-1 font-semibold">
                                     <a
