diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index e1bc7f4..823db71 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -24,20 +24,20 @@
                                 <h4 class="text-[22px] pt-1 font-semibold">
                                     <a
                                         class="hover:text-slate-200"
-                                        href="#"
-                                    >USS LeafyCruiser</a>
+                                        href="{{ path('app_starship_show', { id: ship.id }) }}"
+                                    >{{ ship.name }}</a>
                                 </h4>
                             </div>
                         </div>
                         <div class="flex justify-center min-[1174px]:justify-start mt-2 min-[1174px]:mt-0 shrink-0">
                             <div class="border-r border-white/20 pr-8">
                                 <p class="text-slate-400 text-xs">Captain</p>
-                                <p class="text-xl">Jean-Luc Pickles</p>
+                                <p class="text-xl">{{ ship.captain }}</p>
                             </div>
 
                             <div class="pl-8 w-[100px]">
                                 <p class="text-slate-400 text-xs">Class</p>
-                                <p class="text-xl">Garden</p>
+                                <p class="text-xl">{{ ship.class }}</p>
                             </div>
                         </div>
                     </div>
