diff --git a/config/packages/cache.yaml b/config/packages/cache.yaml
index 7050b11..9069c40 100644
--- a/config/packages/cache.yaml
+++ b/config/packages/cache.yaml
@@ -17,5 +17,5 @@ framework:
         app: cache.adapter.array
 
         # Namespaced pools use the above "app" backend by default
-        #pools:
-            #my.dedicated.cache: null
+        pools:
+            iss_location_pool: null
diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index e48167e..85f7385 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -17,12 +17,12 @@ class MainController extends AbstractController
     public function homepage(
         StarshipRepository $starshipRepository,
         HttpClientInterface $client,
-        CacheInterface $cache,
+        CacheInterface $issLocationPool,
     ): Response {
         $ships = $starshipRepository->findAll();
         $myShip = $ships[array_rand($ships)];
 
-        $issData = $cache->get('iss_location_data', function (ItemInterface $item) use ($client): array {
+        $issData = $issLocationPool->get('iss_location_data', function (ItemInterface $item) use ($client): array {
             $item->expiresAfter(5);
 
             $response = $client->request(Request::METHOD_GET, 'https://api.wheretheiss.at/v1/satellites/25544');
