diff --git a/src/Twig/Extension/AppExtension.php b/src/Twig/Extension/AppExtension.php
new file mode 100644
index 0000000..0525596
--- /dev/null
+++ b/src/Twig/Extension/AppExtension.php
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Twig\Extension;
+
+use App\Twig\Runtime\AppExtensionRuntime;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+use Twig\TwigFunction;
+
+class AppExtension extends AbstractExtension
+{
+    public function getFilters(): array
+    {
+        return [
+            // If your filter generates SAFE HTML, you should add a third
+            // parameter: ['is_safe' => ['html']]
+            // Reference: https://twig.symfony.com/doc/3.x/advanced.html#automatic-escaping
+            new TwigFilter('filter_name', [AppExtensionRuntime::class, 'doSomething']),
+        ];
+    }
+
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('function_name', [AppExtensionRuntime::class, 'doSomething']),
+        ];
+    }
+}
diff --git a/src/Twig/Runtime/AppExtensionRuntime.php b/src/Twig/Runtime/AppExtensionRuntime.php
new file mode 100644
index 0000000..eb9e189
--- /dev/null
+++ b/src/Twig/Runtime/AppExtensionRuntime.php
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Twig\Runtime;
+
+use Twig\Extension\RuntimeExtensionInterface;
+
+class AppExtensionRuntime implements RuntimeExtensionInterface
+{
+    public function __construct()
+    {
+        // Inject dependencies if needed
+    }
+
+    public function doSomething($value)
+    {
+        // ...
+    }
+}
