diff --git a/src/Twig/Runtime/AppExtensionRuntime.php b/src/Twig/Runtime/AppExtensionRuntime.php
index 2fa84a0..c8c8499 100644
--- a/src/Twig/Runtime/AppExtensionRuntime.php
+++ b/src/Twig/Runtime/AppExtensionRuntime.php
@@ -2,19 +2,22 @@
 
 namespace App\Twig\Runtime;
 
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
 use Twig\Extension\RuntimeExtensionInterface;
 
 class AppExtensionRuntime implements RuntimeExtensionInterface
 {
-    public function __construct()
-    {
-        // Inject dependencies if needed
+    public function __construct(
+        private readonly HttpClientInterface $client,
+        private readonly CacheInterface $issLocationPool,
+    ) {
     }
 
     public function getIssLocationData()
     {
-        return $issLocationPool->get('iss_location_data', function () use ($client): array {
-            $response = $client->request('GET', 'https://api.wheretheiss.at/v1/satellites/25544');
+        return $this->issLocationPool->get('iss_location_data', function (): array {
+            $response = $this->client->request('GET', 'https://api.wheretheiss.at/v1/satellites/25544');
 
             return $response->toArray();
         });
