diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 63ded4d..602834e 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -7,6 +7,8 @@ use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 
 class MainController extends AbstractController
@@ -15,12 +17,18 @@ class MainController extends AbstractController
     public function homepage(
         StarshipRepository $starshipRepository,
         HttpClientInterface $client,
+        CacheInterface $cache,
     ): Response {
         $ships = $starshipRepository->findAll();
         $myShip = $ships[array_rand($ships)];
 
-        $response = $client->request('GET', 'https://api.wheretheiss.at/v1/satellites/25544');
-        $issData = $response->toArray();
+        $issData = $cache->get('iss_location_data', function (ItemInterface $item) use ($client): array {
+            $item->expiresAfter(5);
+
+            $response = $client->request('GET', 'https://api.wheretheiss.at/v1/satellites/25544');
+
+            return $response->toArray();
+        });
 
         return $this->render('main/homepage.html.twig', [
             'myShip' => $myShip,
