diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index e48167e..aa0109a 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -19,6 +19,7 @@ class MainController extends AbstractController
         HttpClientInterface $client,
         CacheInterface $cache,
     ): Response {
+        dd($cache);
         $ships = $starshipRepository->findAll();
         $myShip = $ships[array_rand($ships)];
 
