diff --git a/src/Repository/StarshipPartRepository.php b/src/Repository/StarshipPartRepository.php
index 209897f..6a922c7 100644
--- a/src/Repository/StarshipPartRepository.php
+++ b/src/Repository/StarshipPartRepository.php
@@ -4,6 +4,7 @@ namespace App\Repository;
 
 use App\Entity\StarshipPart;
 use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Collections\Collection;
 use Doctrine\Common\Collections\Criteria;
 use Doctrine\Persistence\ManagerRegistry;
 
@@ -22,6 +23,18 @@ class StarshipPartRepository extends ServiceEntityRepository
         return Criteria::create()->andWhere(Criteria::expr()->gt('price', 50000));
     }
 
+    /**
+     * @return Collection<StarshipPart>
+     */
+    public function getExpensiveParts(int $limit = 10): Collection
+    {
+        return $this->createQueryBuilder('sp')
+            ->addCriteria(self::createExpensiveCriteria())
+            ->setMaxResults($limit)
+            ->getQuery()
+            ->getResult();
+    }
+
 //    /**
 //     * @return StarshipPart[] Returns an array of StarshipPart objects
 //     */
