diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index e860b07..da0d78f 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -2,6 +2,7 @@
 
 namespace App\Entity;
 
+use App\Repository\StarshipPartRepository;
 use App\Repository\StarshipRepository;
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
@@ -164,9 +165,7 @@ class Starship
      */
     public function getExpensiveParts(): Collection
     {
-        $criteria = Criteria::create()->andWhere(Criteria::expr()->gt('price', 50000));
-
-        return $this->parts->matching($criteria);
+        return $this->parts->matching(StarshipPartRepository::createExpensiveCriteria());
     }
 
     public function addPart(StarshipPart $part): static
diff --git a/src/Repository/StarshipPartRepository.php b/src/Repository/StarshipPartRepository.php
index 7622388..209897f 100644
--- a/src/Repository/StarshipPartRepository.php
+++ b/src/Repository/StarshipPartRepository.php
@@ -4,6 +4,7 @@ namespace App\Repository;
 
 use App\Entity\StarshipPart;
 use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Collections\Criteria;
 use Doctrine\Persistence\ManagerRegistry;
 
 /**
@@ -16,6 +17,11 @@ class StarshipPartRepository extends ServiceEntityRepository
         parent::__construct($registry, StarshipPart::class);
     }
 
+    public static function createExpensiveCriteria(): Criteria
+    {
+        return Criteria::create()->andWhere(Criteria::expr()->gt('price', 50000));
+    }
+
 //    /**
 //     * @return StarshipPart[] Returns an array of StarshipPart objects
 //     */
