diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 5f42da7..5696315 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -158,6 +158,14 @@ class Starship
         return $this->parts;
     }
 
+    /**
+     * @return Collection<int, StarshipPart>
+     */
+    public function getExpensiveParts(): Collection
+    {
+        return $this->parts;
+    }
+
     public function addPart(StarshipPart $part): static
     {
         if (!$this->parts->contains($part)) {
diff --git a/templates/starship/show.html.twig b/templates/starship/show.html.twig
index 3c67407..50bd21e 100644
--- a/templates/starship/show.html.twig
+++ b/templates/starship/show.html.twig
@@ -57,10 +57,10 @@
                     <p class="text-[22px] font-semibold">{{ ship.arrivedAt|ago }}</p>
 
                     <h4 class="text-xs text-slate-300 font-semibold mt-2 uppercase">
-                        Parts ({{ ship.parts|length }})
+                        Expensive Parts ({{ ship.expensiveParts|length }})
                     </h4>
                     <ul class="text-sm font-medium text-slate-400 tracking-wide">
-                        {% for part in ship.parts %}
+                        {% for part in ship.expensiveParts %}
                             <li class="border-b border-slate-600 py-2">
                                 <span class="block text-white font-semibold">
                                     {{ part.name }} (✦ {{ part.price }})
