diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 1d0e530..e860b07 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -5,6 +5,7 @@ namespace App\Entity;
 use App\Repository\StarshipRepository;
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Collections\Criteria;
 use Doctrine\ORM\Mapping as ORM;
 use Gedmo\Mapping\Annotation\Slug;
 use Gedmo\Mapping\Annotation\Timestampable;
@@ -163,9 +164,9 @@ class Starship
      */
     public function getExpensiveParts(): Collection
     {
-        return $this->parts->filter(function (StarshipPart $part) {
-            return $part->getPrice() > 50000;
-        });
+        $criteria = Criteria::create()->andWhere(Criteria::expr()->gt('price', 50000));
+
+        return $this->parts->matching($criteria);
     }
 
     public function addPart(StarshipPart $part): static
