diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 5696315..1d0e530 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -163,7 +163,9 @@ class Starship
      */
     public function getExpensiveParts(): Collection
     {
-        return $this->parts;
+        return $this->parts->filter(function (StarshipPart $part) {
+            return $part->getPrice() > 50000;
+        });
     }
 
     public function addPart(StarshipPart $part): static
