diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index cada3bf..a0da177 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -27,6 +27,9 @@
                                         href="{{ path('app_starship_show', { id: ship.id }) }}"
                                     >{{ ship.name }}</a>
                                 </h4>
+                                <div>
+                                    Arrived at: {{ ship.arrivedAt }}
+                                </div>
                             </div>
                         </div>
                         <div class="flex justify-center min-[1174px]:justify-start mt-2 min-[1174px]:mt-0 shrink-0">
