diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 19e281a..a4e9d97 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -2,6 +2,7 @@
 
 namespace App\Controller;
 
+use App\Entity\Starship;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
@@ -13,7 +14,11 @@ class MainController extends AbstractController
     public function homepage(
         EntityManagerInterface $em,
     ): Response {
-        $ships = $em->createQuery('SELECT s FROM App\Entity\Starship s')->getResult();
+        $ships = $em->createQueryBuilder()
+            ->select('s')
+            ->from(Starship::class, 's')
+            ->getQuery()
+            ->getResult();
         $myShip = $ships[array_rand($ships)];
 
         return $this->render('main/homepage.html.twig', [
