diff --git a/src/Form/StarshipPartType.php b/src/Form/StarshipPartType.php
new file mode 100644
index 0000000..9a4bdb5
--- /dev/null
+++ b/src/Form/StarshipPartType.php
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Form;
+
+use App\Entity\Starship;
+use App\Entity\StarshipPart;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class StarshipPartType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('name')
+            ->add('price')
+            ->add('notes')
+            ->add('createdAt')
+            ->add('updatedAt')
+            ->add('starship', EntityType::class, [
+                'class' => Starship::class,
+                'choice_label' => 'id',
+            ])
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => StarshipPart::class,
+        ]);
+    }
+}
