diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 2e71809..db4fd60 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -34,5 +34,7 @@ class AppFixtures extends Fixture
             'status' => StarshipStatusEnum::WAITING,
             'arrivedAt' => new \DateTimeImmutable('-1 month'),
         ]);
+
+        StarshipFactory::createMany(20);
     }
 }
