diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 48cb2d4..4d10c30 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -19,11 +19,11 @@ class MainController extends AbstractController
 
         $response = $client->request(Request::METHOD_GET, 'https://api.wheretheiss.at/v1/satellites/25544');
         $issData = $response->toArray();
-        dump($issData);
 
         return $this->render('main/homepage.html.twig', [
             'myShip' => $myShip,
             'ships' => $ships,
+            'issData' => $issData,
         ]);
     }
 }
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 852b457..6f8d044 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -51,6 +51,16 @@
                 Looking for your next galactic ride?
                 <a href="#" class="underline font-semibold">Browse the {{ ships|length * 10 }} starships for sale!</a>
             </p>
+
+            <div>
+                <h2 class="text-4xl font-semibold my-8">ISS Location</h2>
+
+                <p>Time: {{ issData.timestamp|date }}</p>
+                <p>Altitude: {{ issData.altitude }}</p>
+                <p>Latitude: {{ issData.latitude }}</p>
+                <p>Longitude: {{ issData.longitude }}</p>
+                <p>Visibility: {{ issData.visibility }}</p>
+            </div>
         </div>
     </main>
 {% endblock %}
