diff --git a/src/Factory/DroidFactory.php b/src/Factory/DroidFactory.php
index d75ece4..9626faa 100644
--- a/src/Factory/DroidFactory.php
+++ b/src/Factory/DroidFactory.php
@@ -32,8 +32,16 @@ final class DroidFactory extends PersistentProxyObjectFactory
     protected function defaults(): array|callable
     {
         return [
-            'name' => self::faker()->text(255),
-            'primaryFunction' => self::faker()->text(255),
+            'name' => self::faker()->randomElement([
+                'R2-D2', 'C-3PO', 'BB-8', 'ZZZ-123',
+            ]),
+            'primaryFunction' => self::faker()->randomElement([
+                'astromech',
+                'protocol',
+                'astromech',
+                'assassin',
+                'sleeper',
+            ]),
         ];
     }
 
