diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 70bd5ee..369fcc2 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -49,7 +49,7 @@ class AppFixtures extends Fixture
         DroidFactory::createMany(100);
 
         StarshipFactory::createMany(100, fn() => [
-            'droids' => DroidFactory::randomRange(1, 5),
+            //'droids' => DroidFactory::randomRange(1, 5),
         ]);
         StarshipPartFactory::createMany(50);
     }
