diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 09c2f38..2f6aee7 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -3,6 +3,7 @@
 namespace App\DataFixtures;
 
 use App\Entity\Droid;
+use App\Entity\StarshipDroid;
 use App\Entity\StarshipStatusEnum;
 use App\Factory\DroidFactory;
 use App\Factory\StarshipFactory;
@@ -57,6 +58,10 @@ class AppFixtures extends Fixture
 
         $ship = StarshipFactory::random()->_real();
         $droid = DroidFactory::random()->_real();
-        $ship->addDroid($droid);
+        $starshipDroid = new StarshipDroid();
+        $starshipDroid->setStarship($ship);
+        $starshipDroid->setDroid($droid);
+        $manager->persist($starshipDroid);
+        $manager->flush();
     }
 }
