diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 85d3cbf..36ab887 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -13,34 +13,13 @@ class AppFixtures extends Fixture
 {
     public function load(ObjectManager $manager): void
     {
-        $starship = StarshipFactory::createOne([
+        StarshipFactory::createOne([
             'name' => 'USS LeafyCruiser (NCC-0001)',
             'class' => 'Garden',
             'captain' => 'Jean-Luc Pickles',
             'status' => StarshipStatusEnum::IN_PROGRESS,
             'arrivedAt' => new \DateTimeImmutable('-1 day'),
         ]);
-        $droid1 = new Droid();
-        $droid1->setName('IHOP-123');
-        $droid1->setPrimaryFunction('Pancake chef');
-        $starship->addDroid($droid1);
-        $manager->persist($droid1);
-
-        $droid2 = new Droid();
-        $droid2->setName('D-3P0');
-        $droid2->setPrimaryFunction('C-3PO\'s voice coach');
-        $starship->addDroid($droid2);
-        $manager->persist($droid2);
-
-        $droid3 = new Droid();
-        $droid3->setName('BONK-5000');
-        $droid3->setPrimaryFunction('Comedy sidekick');
-        $starship->addDroid($droid3);
-        $manager->persist($droid3);
-        $manager->flush();
-
-        $starship->removeDroid($droid1);
-        $manager->flush();
 
         StarshipFactory::createOne([
             'name' => 'USS Espresso (NCC-1234-C)',
