diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 6df8691..46a235d 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -47,7 +47,7 @@ class AppFixtures extends Fixture
 
         DroidFactory::createMany(100);
 
-        StarshipFactory::createMany(100, [
+        StarshipFactory::createMany(100, fn() => [
             'droids' => DroidFactory::randomRange(1, 5),
         ]);
         StarshipPartFactory::createMany(100);
