diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 29a5e16..5ad1948 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -48,7 +48,7 @@ class AppFixtures extends Fixture
 
         DroidFactory::createMany(100);
 
-        StarshipFactory::createMany(100, [
+        StarshipFactory::createMany(100, fn() => [
             'droids' => DroidFactory::randomRange(1, 5),
         ]);
         StarshipPartFactory::createMany(100);
