diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 0e06c84..2b7d337 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -51,7 +51,7 @@ class Starship
     /**
      * @var Collection<int, StarshipDroid>
      */
-    #[ORM\OneToMany(targetEntity: StarshipDroid::class, mappedBy: 'starship')]
+    #[ORM\OneToMany(targetEntity: StarshipDroid::class, mappedBy: 'starship', cascade: ['persist'])]
     private Collection $starshipDroids;
 
     public function __construct()
