diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index d7ca046..c0fe7a8 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -49,13 +49,8 @@ class AppFixtures extends Fixture
         DroidFactory::createMany(100);
 
         StarshipFactory::createMany(100, fn() => [
-            //'droids' => DroidFactory::randomRange(1, 5),
+            'droids' => DroidFactory::randomRange(1, 5),
         ]);
         StarshipPartFactory::createMany(100);
-
-        $ship = StarshipFactory::random()->_real();
-        $droid = DroidFactory::random()->_real();
-        $ship->addDroid($droid);
-        $manager->flush();
     }
 }
