diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index bfdbe0e..cf21c0d 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -44,5 +44,9 @@ class AppFixtures extends Fixture
             //'droids' => DroidFactory::randomRange(1, 5),
         ]);
         StarshipPartFactory::createMany(100);
+
+        $ship = StarshipFactory::random()->_real();
+        $droid = DroidFactory::random()->_real();
+        $ship->addDroid($droid);
     }
 }
