diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index ff6e4ae..85d3cbf 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -39,6 +39,9 @@ class AppFixtures extends Fixture
         $manager->persist($droid3);
         $manager->flush();
 
+        $starship->removeDroid($droid1);
+        $manager->flush();
+
         StarshipFactory::createOne([
             'name' => 'USS Espresso (NCC-1234-C)',
             'class' => 'Latte',
