diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 4514d89..948b8ea 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -26,7 +26,7 @@ class StarshipRepository extends ServiceEntityRepository
     {
         $query = $this->createQueryBuilder('s')
             ->andWhere('s.status != :status')
-            ->orderBy('s.arrivedAt', 'DESC')
+            ->orderBy('COUNT(droid)', 'ASC')
             ->leftJoin('s.droids', 'droid')
             ->groupBy('s.id')
             ->setParameter('status', StarshipStatusEnum::COMPLETED)
