diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 1d725e4..fa98e33 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -223,7 +223,7 @@ class Starship
 
     public function getDroidNames(): string
     {
-        return implode(', ', $this->droids->map(fn(Droid $droid) => $droid->getName())->toArray());
+        return implode(', ', $this->getDroids()->map(fn(Droid $droid) => $droid->getName())->toArray());
     }
 
     /**
