diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 83a3506..09c2f38 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -54,5 +54,9 @@ class AppFixtures extends Fixture
             //'droids' => DroidFactory::randomRange(1, 5),
         ]);
         StarshipPartFactory::createMany(100);
+
+        $ship = StarshipFactory::random()->_real();
+        $droid = DroidFactory::random()->_real();
+        $ship->addDroid($droid);
     }
 }
