diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index fa98e33..74dcbb6 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -202,7 +202,7 @@ class Starship
      */
     public function getDroids(): Collection
     {
-        return $this->droids;
+        return $this->starshipDroids->map(fn (StarshipDroid $starshipDroid) => $starshipDroid->getDroid());
     }
 
     public function addDroid(Droid $droid): static
