diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 6b49a19..4514d89 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -27,6 +27,8 @@ class StarshipRepository extends ServiceEntityRepository
         $query = $this->createQueryBuilder('s')
             ->andWhere('s.status != :status')
             ->orderBy('s.arrivedAt', 'DESC')
+            ->leftJoin('s.droids', 'droid')
+            ->groupBy('s.id')
             ->setParameter('status', StarshipStatusEnum::COMPLETED)
             ->getQuery()
         ;
