diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 46a235d..14b48f7 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -48,7 +48,7 @@ class AppFixtures extends Fixture
         DroidFactory::createMany(100);
 
         StarshipFactory::createMany(100, fn() => [
-            'droids' => DroidFactory::randomRange(1, 5),
+            //'droids' => DroidFactory::randomRange(1, 5),
         ]);
         StarshipPartFactory::createMany(100);
     }
