diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 948b8ea..edc4c1f 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -26,8 +26,8 @@ class StarshipRepository extends ServiceEntityRepository
     {
         $query = $this->createQueryBuilder('s')
             ->andWhere('s.status != :status')
-            ->orderBy('COUNT(droid)', 'ASC')
-            ->leftJoin('s.droids', 'droid')
+            ->orderBy('COUNT(starshipDroid)', 'ASC')
+            ->leftJoin('s.starshipDroids', 'starshipDroid')
             ->groupBy('s.id')
             ->setParameter('status', StarshipStatusEnum::COMPLETED)
             ->getQuery()
