diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 8d6574e..d7ca046 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -55,10 +55,7 @@ class AppFixtures extends Fixture
 
         $ship = StarshipFactory::random()->_real();
         $droid = DroidFactory::random()->_real();
-        $starshipDroid = new StarshipDroid();
-        $starshipDroid->setStarship($ship);
-        $starshipDroid->setDroid($droid);
-        $manager->persist($starshipDroid);
+        $ship->addDroid($droid);
         $manager->flush();
     }
 }
