diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 88b190a..339a4ee 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -220,4 +220,9 @@ class Starship
 
         return $this;
     }
+
+    public function getDroidNames(): string
+    {
+        return implode(', ', $this->droids->map(fn(Droid $droid) => $droid->getName())->toArray());
+    }
 }
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 367be19..08cee84 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -35,7 +35,9 @@
                                     Arrived: {{ ship.arrivedAt|ago }}
                                 </div>
                                 <div>
-                                    Parts: {{ ship.parts|length }}
+                                    Parts: {{ ship.parts|length }}</div>
+                                <div>
+                                    Droids: {{ ship.droidNames ?: 'none' }}
                                 </div>
                             </div>
                         </div>
