diff --git a/src/Entity/Droid.php b/src/Entity/Droid.php
index 3a1cad3..da39b5e 100644
--- a/src/Entity/Droid.php
+++ b/src/Entity/Droid.php
@@ -21,15 +21,8 @@ class Droid
     #[ORM\Column(length: 255)]
     private ?string $primaryFunction = null;
 
-    /**
-     * @var Collection<int, Starship>
-     */
-    #[ORM\ManyToMany(targetEntity: Starship::class, mappedBy: 'droids')]
-    private Collection $starships;
-
     public function __construct()
     {
-        $this->starships = new ArrayCollection();
     }
 
     public function getId(): ?int
diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 339a4ee..36209a3 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -48,16 +48,9 @@ class Starship
     #[ORM\OrderBy(['name' => 'ASC'])]
     private Collection $parts;
 
-    /**
-     * @var Collection<int, Droid>
-     */
-    #[ORM\ManyToMany(targetEntity: Droid::class, inversedBy: 'starships')]
-    private Collection $droids;
-
     public function __construct()
     {
         $this->parts = new ArrayCollection();
-        $this->droids = new ArrayCollection();
     }
 
     public function getId(): ?int
