diff --git a/src/Entity/StarshipDroid.php b/src/Entity/StarshipDroid.php
index 0b427a3..5befc36 100644
--- a/src/Entity/StarshipDroid.php
+++ b/src/Entity/StarshipDroid.php
@@ -24,6 +24,11 @@ class StarshipDroid
     #[ORM\JoinColumn(nullable: false)]
     private ?Starship $starship = null;
 
+    public function __construct()
+    {
+        $this->assignedAt = new \DateTimeImmutable();
+    }
+
     public function getId(): ?int
     {
         return $this->id;
