diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 9470f12..cfdda31 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -15,7 +15,7 @@ class MainController extends AbstractController
         StarshipRepository $repository,
         Request $request,
     ): Response {
-        $ships = $repository->findIncomplete();
+        $ships = $repository->findIncompleteOrderedByDroidCount();
         $ships->setMaxPerPage(5);
         $ships->setCurrentPage($request->query->get('page', 1));
 
diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 8afb7ec..6b49a19 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -22,7 +22,7 @@ class StarshipRepository extends ServiceEntityRepository
     /**
      * @return Starship[]
      */
-    public function findIncomplete(): Pagerfanta
+    public function findIncompleteOrderedByDroidCount(): Pagerfanta
     {
         $query = $this->createQueryBuilder('s')
             ->andWhere('s.status != :status')
