diff --git a/src/Controller/AdminController.php b/src/Controller/AdminController.php
index fb1d020..802e96b 100644
--- a/src/Controller/AdminController.php
+++ b/src/Controller/AdminController.php
@@ -6,6 +6,7 @@ use App\Entity\StarshipPart;
 use App\Form\StarshipPartType;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\SubmitButton;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
@@ -28,8 +29,8 @@ class AdminController extends AbstractController
 
             $this->addFlash('success', sprintf('The part "%s" was successfully created.', $part->getName()));
 
+            /** @var SubmitButton $createAndAddNewBtn */
             $createAndAddNewBtn = $form->get('createAndAddNew');
-            dd($createAndAddNewBtn);
 
             return $this->redirectToRoute('app_part_index');
         }
