diff --git a/src/Controller/PartController.php b/src/Controller/PartController.php
index 25e775c..a0faab4 100644
--- a/src/Controller/PartController.php
+++ b/src/Controller/PartController.php
@@ -12,7 +12,7 @@ final class PartController extends AbstractController
     #[Route('/parts', name: 'app_part_index')]
     public function index(StarshipPartRepository $repository): Response
     {
-        $parts = $repository->findAll();
+        $parts = $repository->findAllOrderedByPrice();
 
         return $this->render('part/index.html.twig', [
             'parts' => $parts,
diff --git a/src/Repository/StarshipPartRepository.php b/src/Repository/StarshipPartRepository.php
index 6a922c7..d46a12d 100644
--- a/src/Repository/StarshipPartRepository.php
+++ b/src/Repository/StarshipPartRepository.php
@@ -35,28 +35,14 @@ class StarshipPartRepository extends ServiceEntityRepository
             ->getResult();
     }
 
-//    /**
-//     * @return StarshipPart[] Returns an array of StarshipPart objects
-//     */
-//    public function findByExampleField($value): array
-//    {
-//        return $this->createQueryBuilder('s')
-//            ->andWhere('s.exampleField = :val')
-//            ->setParameter('val', $value)
-//            ->orderBy('s.id', 'ASC')
-//            ->setMaxResults(10)
-//            ->getQuery()
-//            ->getResult()
-//        ;
-//    }
-
-//    public function findOneBySomeField($value): ?StarshipPart
-//    {
-//        return $this->createQueryBuilder('s')
-//            ->andWhere('s.exampleField = :val')
-//            ->setParameter('val', $value)
-//            ->getQuery()
-//            ->getOneOrNullResult()
-//        ;
-//    }
+    /**
+     * @return array<StarshipPart>
+     */
+    public function findAllOrderedByPrice(): array
+    {
+        return $this->createQueryBuilder('sp')
+            ->orderBy('sp.price', 'DESC')
+            ->getQuery()
+            ->getResult();
+    }
 }
