diff --git a/templates/part/index.html.twig b/templates/part/index.html.twig
index 5c2fb8f..776dde4 100644
--- a/templates/part/index.html.twig
+++ b/templates/part/index.html.twig
@@ -13,7 +13,7 @@
                     <div class="ml-5">
                         <h4 class="text-[22px] font-semibold">
                             <a class="hover:text-slate-200" href="#">
-                                {{ part.name }} <span class="text-sm text-slate-400">(assigned to SHIP NAME)</span>
+                                {{ part.name }} <span class="text-sm text-slate-400">(assigned to {{ part.starship.name }})</span>
                             </a>
                         </h4>
                         <div class="text-lg text-green-400 font-medium">✦{{ part.price }}</div>
