diff --git a/src/Repository/StarshipPartRepository.php b/src/Repository/StarshipPartRepository.php
index c8b54d5..679e3c0 100644
--- a/src/Repository/StarshipPartRepository.php
+++ b/src/Repository/StarshipPartRepository.php
@@ -43,6 +43,7 @@ class StarshipPartRepository extends ServiceEntityRepository
         return $this->createQueryBuilder('sp')
             ->orderBy('sp.price', 'DESC')
             ->innerJoin('sp.starship', 's')
+            ->addSelect('s')
             ->getQuery()
             ->getResult();
     }
