diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 6947be0..8f887b6 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -3,6 +3,7 @@
 namespace App\Controller;
 
 use App\Repository\StarshipRepository;
+use Symfony\Bridge\Twig\Command\DebugCommand;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
 use Symfony\Component\HttpFoundation\Request;
@@ -19,6 +20,7 @@ class MainController extends AbstractController
         StarshipRepository $starshipRepository,
         HttpClientInterface $client,
         CacheInterface $issLocationPool,
+        DebugCommand $twigDebugCommand,
     ): Response {
         $ships = $starshipRepository->findAll();
         $myShip = $ships[array_rand($ships)];
