diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 8f887b6..fd35998 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -20,6 +20,7 @@ class MainController extends AbstractController
         StarshipRepository $starshipRepository,
         HttpClientInterface $client,
         CacheInterface $issLocationPool,
+        #[Autowire(service: 'twig.command.debug')]
         DebugCommand $twigDebugCommand,
     ): Response {
         $ships = $starshipRepository->findAll();
