diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 4a8a659..f976f7d 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -22,6 +22,7 @@ class MainController extends AbstractController
         CacheInterface $issLocationPool,
         #[Autowire('%iss_location_cache_ttl%')]
         $issLocationCacheTtl,
+        #[Autowire('@twig.command.debug')]
         DebugCommand $twigDebugCommand,
     ): Response {
         //dd($issLocationCacheTtl);
