diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 17b2993..5f42da7 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -42,7 +42,7 @@ class Starship
     /**
      * @var Collection<int, StarshipPart>
      */
-    #[ORM\OneToMany(targetEntity: StarshipPart::class, mappedBy: 'starship', orphanRemoval: true)]
+    #[ORM\OneToMany(targetEntity: StarshipPart::class, mappedBy: 'starship', fetch: 'EXTRA_LAZY', orphanRemoval: true)]
     #[ORM\OrderBy(['name' => 'ASC'])]
     private Collection $parts;
 
