diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 57b5b5c..3d3419c 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -42,7 +42,7 @@ class Starship
     /**
      * @var Collection<int, StarshipPart>
      */
-    #[ORM\OneToMany(targetEntity: StarshipPart::class, mappedBy: 'starship')]
+    #[ORM\OneToMany(targetEntity: StarshipPart::class, mappedBy: 'starship', fetch: 'EXTRA_LAZY')]
     #[ORM\OrderBy(['name' => 'ASC'])]
     private Collection $parts;
 
