diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index a36bb59..367be19 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -34,6 +34,9 @@
                                 <div>
                                     Arrived: {{ ship.arrivedAt|ago }}
                                 </div>
+                                <div>
+                                    Parts: {{ ship.parts|length }}
+                                </div>
                             </div>
                         </div>
                         <div class="flex justify-center min-[1174px]:justify-start mt-2 min-[1174px]:mt-0 shrink-0">
