diff --git a/templates/admin/starship-part/new.html.twig b/templates/admin/starship-part/new.html.twig
index f453852..71018b9 100644
--- a/templates/admin/starship-part/new.html.twig
+++ b/templates/admin/starship-part/new.html.twig
@@ -17,7 +17,7 @@
 
             {{ form_row(form.notes) }}
 
-            {{ form_row(form.createAndAddNew) }}
+            {{ form_widget(form.createAndAddNew) }}
             <button type="submit" class="text-white bg-green-700 hover:bg-green-800 rounded-lg px-5 py-2.5 me-2 mb-2 cursor-pointer">Create and close</button>
 
             {{ form_rest(form) }}
