diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index 0cc27fb..17b2993 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -43,6 +43,7 @@ class Starship
      * @var Collection<int, StarshipPart>
      */
     #[ORM\OneToMany(targetEntity: StarshipPart::class, mappedBy: 'starship', orphanRemoval: true)]
+    #[ORM\OrderBy(['name' => 'ASC'])]
     private Collection $parts;
 
     public function __construct()
