diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index a6f8c7e..3f241d0 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -36,6 +36,12 @@ class AppFixtures extends Fixture
             'arrivedAt' => new \DateTimeImmutable('-1 month'),
         ]);
 
+        $starshipPart = StarshipPartFactory::createOne([
+            'name' => 'Toilet Paper',
+            'starship' => $ship,
+        ]);
+        dump($starshipPart);
+
         StarshipFactory::createMany(20);
         StarshipPartFactory::createMany(100);
     }
