diff --git a/src/Entity/Starship.php b/src/Entity/Starship.php
index d150913..0cc27fb 100644
--- a/src/Entity/Starship.php
+++ b/src/Entity/Starship.php
@@ -42,7 +42,7 @@ class Starship
     /**
      * @var Collection<int, StarshipPart>
      */
-    #[ORM\OneToMany(targetEntity: StarshipPart::class, mappedBy: 'starship')]
+    #[ORM\OneToMany(targetEntity: StarshipPart::class, mappedBy: 'starship', orphanRemoval: true)]
     private Collection $parts;
 
     public function __construct()
