diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 817af2d..c6b3d6d 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -42,7 +42,6 @@ class AppFixtures extends Fixture
         ])->_real();
         $ship->removePart($starshipPart);
         $manager->flush();
-        dump($starshipPart);
 
         StarshipFactory::createMany(20);
         StarshipPartFactory::createMany(100);
