diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 3f241d0..c93eb82 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -34,12 +34,12 @@ class AppFixtures extends Fixture
             'captain' => 'Kathryn Journeyway',
             'status' => StarshipStatusEnum::WAITING,
             'arrivedAt' => new \DateTimeImmutable('-1 month'),
-        ]);
+        ])->_real();
 
         $starshipPart = StarshipPartFactory::createOne([
             'name' => 'Toilet Paper',
             'starship' => $ship,
-        ]);
+        ])->_real();
         dump($starshipPart);
 
         StarshipFactory::createMany(20);
