diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index ef1da9d..cb16013 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -13,6 +13,17 @@ class AppFixtures extends Fixture
 {
     public function load(ObjectManager $manager): void
     {
+        $starship = StarshipFactory::createOne();
+        $part1 = new StarshipPart();
+        $part1->setName('Warp Core');
+        $part1->setPrice(1000);
+        $part2 = new StarshipPart();
+        $part2->setName('Phaser Array');
+        $part2->setPrice(500);
+        $manager->persist($part1);
+        $manager->persist($part2);
+
+        $manager->flush();
         StarshipFactory::createOne([
             'name' => 'USS LeafyCruiser (NCC-0001)',
             'class' => 'Garden',
