diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index cb16013..1176afa 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -13,7 +13,7 @@ class AppFixtures extends Fixture
 {
     public function load(ObjectManager $manager): void
     {
-        $starship = StarshipFactory::createOne();
+        $starship = StarshipFactory::createOne()->_real();
         $part1 = new StarshipPart();
         $part1->setName('Warp Core');
         $part1->setPrice(1000);
