diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 821fc41..775bb31 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -4,6 +4,7 @@ namespace App\Controller;
 
 use App\Repository\StarshipRepository;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
 
@@ -12,10 +13,11 @@ class MainController extends AbstractController
     #[Route('/', name: 'app_homepage')]
     public function homepage(
         StarshipRepository $repository,
+        Request $request,
     ): Response {
         $ships = $repository->findIncomplete();
         $ships->setMaxPerPage(5);
-        $ships->setCurrentPage(2);
+        $ships->setCurrentPage($request->query->getInt('page', 1));
 
         $myShip = $repository->findMyShip();
 
