diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index a3802b7..f251d8a 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -19,7 +19,7 @@ class MainController extends AbstractController
         HttpClientInterface $client,
         CacheInterface $issLocationPool,
     ): Response {
-        dd($this->getParameter('kernel.project_dir'));
+        dd($this->getParameter('iss_location_cache_ttl'));
         $ships = $starshipRepository->findAll();
         $myShip = $ships[array_rand($ships)];
 
