diff --git a/src/Controller/PartController.php b/src/Controller/PartController.php
index 0dfc8e3..5b49e9a 100644
--- a/src/Controller/PartController.php
+++ b/src/Controller/PartController.php
@@ -8,7 +8,7 @@ use Symfony\Component\Routing\Attribute\Route;
 
 final class PartController extends AbstractController
 {
-    #[Route('/part', name: 'app_part')]
+    #[Route('/parts', name: 'app_part_index')]
     public function index(): Response
     {
         return $this->render('part/index.html.twig', [
diff --git a/templates/base.html.twig b/templates/base.html.twig
index f471ee5..6d7696b 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -22,8 +22,8 @@
                         <a class="hover:text-amber-400 pt-2" href="{{ path('app_homepage') }}">
                             Home
                         </a>
-                        <a class="hover:text-amber-400  pt-2" href="#">
-                            About
+                        <a class="hover:text-amber-400  pt-2" href="{{ path('app_part_index') }}">
+                            Parts
                         </a>
                         <a class="hover:text-amber-400 pt-2" href="#">
                             Contact
diff --git a/templates/part/index.html.twig b/templates/part/index.html.twig
index 64f5e5f..14f25a7 100644
--- a/templates/part/index.html.twig
+++ b/templates/part/index.html.twig
@@ -1,6 +1,6 @@
 {% extends 'base.html.twig' %}
 
-{% block title %}Hello PartController!{% endblock %}
+{% block title %}Parts!{% endblock %}
 
 {% block body %}
 <style>
