diff --git a/src/Controller/PartController.php b/src/Controller/PartController.php
new file mode 100644
index 0000000..0dfc8e3
--- /dev/null
+++ b/src/Controller/PartController.php
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+final class PartController extends AbstractController
+{
+    #[Route('/part', name: 'app_part')]
+    public function index(): Response
+    {
+        return $this->render('part/index.html.twig', [
+            'controller_name' => 'PartController',
+        ]);
+    }
+}
diff --git a/templates/part/index.html.twig b/templates/part/index.html.twig
new file mode 100644
index 0000000..64f5e5f
--- /dev/null
+++ b/templates/part/index.html.twig
@@ -0,0 +1,20 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Hello PartController!{% endblock %}
+
+{% block body %}
+<style>
+    .example-wrapper { margin: 1em auto; max-width: 800px; width: 95%; font: 18px/1.5 sans-serif; }
+    .example-wrapper code { background: #F5F5F5; padding: 2px 6px; }
+</style>
+
+<div class="example-wrapper">
+    <h1>Hello {{ controller_name }}! ✅</h1>
+
+    This friendly message is coming from:
+    <ul>
+        <li>Your controller at <code>/Users/weaverryan/Sites/knp/knpu-repos/symfony7-ep4/src/Controller/PartController.php</code></li>
+        <li>Your template at <code>/Users/weaverryan/Sites/knp/knpu-repos/symfony7-ep4/templates/part/index.html.twig</code></li>
+    </ul>
+</div>
+{% endblock %}
