diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index f57dff8..db4fd60 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -2,7 +2,6 @@
 
 namespace App\DataFixtures;
 
-use App\Entity\Starship;
 use App\Entity\StarshipStatusEnum;
 use App\Factory\StarshipFactory;
 use Doctrine\Bundle\FixturesBundle\Fixture;
@@ -12,17 +11,6 @@ class AppFixtures extends Fixture
 {
     public function load(ObjectManager $manager): void
     {
-        $ship = new Starship(
-            'USS LeafyCruiser (NCC-0001)',
-            'Garden',
-            'Jean-Luc Pickles',
-        );
-
-        $manager->persist($ship);
-        $manager->flush();
-
-        return;
-
         StarshipFactory::createOne([
             'name' => 'USS LeafyCruiser (NCC-0001)',
             'class' => 'Garden',
