diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index b4dd0ae..1c56951 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -25,6 +25,10 @@ class StarshipApiController extends AbstractController
     {
         $starship = $repository->find($id);
 
+        if (!$starship) {
+            throw $this->createNotFoundException('Starship not found');
+        }
+
         return $this->json($starship);
     }
 }
