diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 8a38153..b4dd0ae 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -21,8 +21,10 @@ class StarshipApiController extends AbstractController
     }
 
     #[Route('/{id<\d+>}', methods: ['GET'])]
-    public function get(int $id): Response
+    public function get(int $id, StarshipRepository $repository): Response
     {
-        dd($id);
+        $starship = $repository->find($id);
+
+        return $this->json($starship);
     }
 }
diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index c083a8d..f6e0f83 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -42,4 +42,15 @@ class StarshipRepository extends AbstractController
             ),
         ];
     }
+
+    public function find(int $id): ?Starship
+    {
+        foreach ($this->findAll() as $starship) {
+            if ($starship->getId() === $id) {
+                return $starship;
+            }
+        }
+
+        return null;
+    }
 }
