diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index e50f16e..7833cfa 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -19,8 +19,10 @@ class StarshipApiController extends AbstractController
     }
 
     #[Route('/{id<\d+>}', methods: ['GET'])]
-    public function get(int $id): Response
+    public function get(int $id, StarshipRepository $repository): Response
     {
-        dd($id);
+        $starship = $repository->find($id);
+
+        return $this->json($starship);
     }
 }
diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index 515a46c..382aa35 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -39,4 +39,15 @@ class StarshipRepository
             ),
         ];
     }
+
+    public function find(int $id): ?Starship
+    {
+        foreach ($this->findAll() as $starship) {
+            if ($starship->getId() === $id) {
+                return $starship;
+            }
+        }
+
+        return null;
+    }
 }
