diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 34e8a9c..07d05a7 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -24,7 +24,11 @@
     <table>
         <tr>
             <th>Name</th>
-            <td>{{ myShip.name }}</td>
+            <td>
+                <a href="{{ path('app_starship_show', {
+                    id: myShip.id
+                }) }}">{{ myShip.name }}</a>
+            </td>
         </tr>
         <tr>
             <th>Class</th>
