diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 5c0ee17..cc7b805 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -17,8 +17,8 @@ class StarshipApiController extends AbstractController
         return $this->json($starships);
     }
 
-    #[Route('/api/starships/{id}')]
-    public function get($id): Response
+    #[Route('/api/starships/{id<\d+>}')]
+    public function get(int $id): Response
     {
         dd($id);
     }
