diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 5b923f8..466f1a1 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -24,7 +24,11 @@
     <table>
         <tr>
             <th>Name</th>
-            <td>{{ featuredShip.name }}</td>
+            <td>
+                <a href="{{ path('app_starship_show', {
+                    id: featuredShip.id
+                }) }}">{{ featuredShip.name }}</a>
+            </td>
         </tr>
         <tr>
             <th>Class</th>
