diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 833116f..f272f08 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -19,8 +19,8 @@ class StarshipApiController extends AbstractController
         return $this->json($starships);
     }
 
-    #[Route('/api/starships/{id}')]
-    public function get($id): Response
+    #[Route('/api/starships/{id<\d+>}')]
+    public function get(int $id): Response
     {
         dd($id);
     }
