diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 7833cfa..381068f 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -23,6 +23,10 @@ class StarshipApiController extends AbstractController
     {
         $starship = $repository->find($id);
 
+        if (!$starship) {
+            throw $this->createNotFoundException('Starship not found');
+        }
+
         return $this->json($starship);
     }
 }
