diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 9cebecd..833116f 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -18,4 +18,10 @@ class StarshipApiController extends AbstractController
 
         return $this->json($starships);
     }
+
+    #[Route('/api/starships/{id}')]
+    public function get($id): Response
+    {
+        dd($id);
+    }
 }
