diff --git a/templates/part/index.html.twig b/templates/part/index.html.twig
index bd622e5..5e1bc34 100644
--- a/templates/part/index.html.twig
+++ b/templates/part/index.html.twig
@@ -10,6 +10,7 @@
                 <input type="text"
                        placeholder="Search..."
                        name="query"
+                       value="{{ app.request.query.get('query') }}"
                        class="w-full p-3 pl-10 border border-gray-300 rounded-lg shadow-sm focus:ring-2 focus:ring-blue-500 focus:border-blue-500"
                 >
                 <svg class="absolute left-3 top-3 w-5 h-5 text-gray-400" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
