diff --git a/templates/part/index.html.twig b/templates/part/index.html.twig
index 776dde4..3fecb81 100644
--- a/templates/part/index.html.twig
+++ b/templates/part/index.html.twig
@@ -3,6 +3,19 @@
 {% block title %}Parts!{% endblock %}
 
 {% block body %}
+
+    <div class="flex justify-end mt-6 mb-6">
+        <div class="relative w-full max-w-md">
+            <input type="text"
+                   placeholder="Search..."
+                   class="w-full p-3 pl-10 border border-gray-300 rounded-lg shadow-sm focus:ring-2 focus:ring-blue-500 focus:border-blue-500"
+            >
+            <svg class="absolute left-3 top-3 w-5 h-5 text-gray-400" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-4.35-4.35m0 0A8.5 8.5 0 1011 19.5a8.5 8.5 0 005.65-2.85z" />
+            </svg>
+        </div>
+    </div>
+
     <div class="space-y-5 mx-5">
         {% for part in parts %}
             <div class="bg-[#16202A] rounded-2xl p-5 flex flex-col min-[1174px]:flex-row min-[1174px]:justify-between items-center">
