diff --git a/templates/part/index.html.twig b/templates/part/index.html.twig
index a81bf8e..3e72936 100644
--- a/templates/part/index.html.twig
+++ b/templates/part/index.html.twig
@@ -17,6 +17,8 @@
                     <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-4.35-4.35m0 0A8.5 8.5 0 1011 19.5a8.5 8.5 0 005.65-2.85z" />
                 </svg>
             </form>
+
+            {{ form(searchForm) }}
         </div>
     </div>
 
