diff --git a/src/Repository/StarshipPartRepository.php b/src/Repository/StarshipPartRepository.php
index a836f3d..8a3ff80 100644
--- a/src/Repository/StarshipPartRepository.php
+++ b/src/Repository/StarshipPartRepository.php
@@ -47,8 +47,8 @@ class StarshipPartRepository extends ServiceEntityRepository
         ;
 
         if ($search) {
-            $qb->andWhere('sp.name LIKE :search')
-                ->setParameter('search', '%'.$search.'%');
+            $qb->andWhere('LOWER(sp.name) LIKE :search')
+                ->setParameter('search', '%'.strtolower($search).'%');
         }
 
         return $qb->getQuery()
