diff --git a/src/Repository/StarshipPartRepository.php b/src/Repository/StarshipPartRepository.php
index 8a3ff80..d429f55 100644
--- a/src/Repository/StarshipPartRepository.php
+++ b/src/Repository/StarshipPartRepository.php
@@ -47,8 +47,8 @@ class StarshipPartRepository extends ServiceEntityRepository
         ;
 
         if ($search) {
-            $qb->andWhere('LOWER(sp.name) LIKE :search')
-                ->setParameter('search', '%'.strtolower($search).'%');
+            $qb->andWhere('LOWER(sp.name) LIKE :search OR LOWER(sp.notes) LIKE :search')
+                ->setParameter('search', '%'.$search.'%');
         }
 
         return $qb->getQuery()
