diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 70187a4..e8983fc 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -13,12 +13,11 @@ class MainController extends AbstractController
     public function homepage(StarshipRepository $starshipRepository): Response
     {
         $ships = $starshipRepository->findAll();
-        $starshipCount = count($ships);
         $myShip = $ships[array_rand($ships)];
 
         return $this->render('main/homepage.html.twig', [
-            'numberOfStarships' => $starshipCount,
             'myShip' => $myShip,
+            'ships' => $ships,
         ]);
     }
 }
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index f056988..34e8a9c 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -8,9 +8,9 @@
 </h1>
 
 <div>
-    Browse through {{ numberOfStarships * 10 }} starships!
+    Browse through {{ ships|length * 10 }} starships!
 
-    {% if numberOfStarships > 400 %}
+    {% if ships|length > 2 %}
         <p>
             {# Do you think "shiploads" will pass the legal team? #}
             That's a shiploads of ships!
