diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 42d1b97..3815c7c 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -3,6 +3,7 @@
 namespace App\Controller;
 
 use App\Model\Starship;
+use Psr\Log\LoggerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
@@ -10,8 +11,9 @@ use Symfony\Component\Routing\Attribute\Route;
 class StarshipApiController extends AbstractController
 {
     #[Route('/api/starships')]
-    public function getCollection(): Response
+    public function getCollection(LoggerInterface $logger): Response
     {
+        dd($logger);
         $starships = [
             new Starship(
                 1,
