diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 826fa48..166ff05 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -3,7 +3,6 @@
 namespace App\Controller;
 
 use App\Repository\StarshipRepository;
-use Psr\Log\LoggerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
@@ -11,9 +10,8 @@ use Symfony\Component\Routing\Attribute\Route;
 class StarshipApiController extends AbstractController
 {
     #[Route('/api/starships')]
-    public function getCollection(LoggerInterface $logger, StarshipRepository $repository): Response
+    public function getCollection(StarshipRepository $repository): Response
     {
-        $logger->info('Starship collection retrieved');
         $starships = $repository->findAll();
 
         return $this->json($starships);
diff --git a/src/Repository/StarshipRepository.php b/src/Repository/StarshipRepository.php
index a9b483b..515a46c 100644
--- a/src/Repository/StarshipRepository.php
+++ b/src/Repository/StarshipRepository.php
@@ -3,11 +3,18 @@
 namespace App\Repository;
 
 use App\Model\Starship;
+use Psr\Log\LoggerInterface;
 
 class StarshipRepository
 {
+    public function __construct(private LoggerInterface $logger)
+    {
+    }
+
     public function findAll(): array
     {
+        $this->logger->info('Starship collection retrieved');
+
         return [
             new Starship(
                 1,
