diff --git a/src/Controller/StarshipApiController.php b/src/Controller/StarshipApiController.php
index 3815c7c..3aa1699 100644
--- a/src/Controller/StarshipApiController.php
+++ b/src/Controller/StarshipApiController.php
@@ -13,7 +13,7 @@ class StarshipApiController extends AbstractController
     #[Route('/api/starships')]
     public function getCollection(LoggerInterface $logger): Response
     {
-        dd($logger);
+        $logger->info('Starship collection retrieved');
         $starships = [
             new Starship(
                 1,
