diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 312ee70..4959230 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -34,6 +34,7 @@ class AppFixtures extends Fixture
         $part->setNotes('There\'s no air drag in space, but it looks cool.');
         $part->setPrice(500);
         $manager->persist($part);
+        $part->setStarship($starship);
         $manager->flush();
 
         StarshipFactory::createOne([
