diff --git a/src/Factory/StarshipPartFactory.php b/src/Factory/StarshipPartFactory.php
index f4249c5..4922221 100644
--- a/src/Factory/StarshipPartFactory.php
+++ b/src/Factory/StarshipPartFactory.php
@@ -51,7 +51,7 @@ final class StarshipPartFactory extends PersistentProxyObjectFactory
             'name' => $randomPart[0],
             'notes' => $randomPart[1],
             'price' => self::faker()->randomNumber(5),
-            'starship' => StarshipFactory::createOne(),
+            'starship' => StarshipFactory::new(),
         ];
     }
 
