diff --git a/src/Factory/StarshipPartFactory.php b/src/Factory/StarshipPartFactory.php
index 67d4275..3d9875f 100644
--- a/src/Factory/StarshipPartFactory.php
+++ b/src/Factory/StarshipPartFactory.php
@@ -52,7 +52,7 @@ final class StarshipPartFactory extends PersistentProxyObjectFactory
             'name' => $randomPart[0],
             'notes' => $randomPart[1],
             'price' => self::faker()->randomNumber(5),
-            'starship' => StarshipFactory::randomOrCreate([
+            'starship' => StarshipFactory::createOne([
                 'status' => StarshipStatusEnum::IN_PROGRESS,
             ]),
         ];
