diff --git a/src/Factory/StarshipPartFactory.php b/src/Factory/StarshipPartFactory.php
index 8c1da12..67d4275 100644
--- a/src/Factory/StarshipPartFactory.php
+++ b/src/Factory/StarshipPartFactory.php
@@ -3,6 +3,7 @@
 namespace App\Factory;
 
 use App\Entity\StarshipPart;
+use App\Entity\StarshipStatusEnum;
 use Zenstruck\Foundry\Persistence\PersistentProxyObjectFactory;
 
 /**
@@ -51,6 +52,9 @@ final class StarshipPartFactory extends PersistentProxyObjectFactory
             'name' => $randomPart[0],
             'notes' => $randomPart[1],
             'price' => self::faker()->randomNumber(5),
+            'starship' => StarshipFactory::randomOrCreate([
+                'status' => StarshipStatusEnum::IN_PROGRESS,
+            ]),
         ];
     }
 
