diff --git a/src/Factory/StarshipPartFactory.php b/src/Factory/StarshipPartFactory.php
new file mode 100644
index 0000000..00f6e15
--- /dev/null
+++ b/src/Factory/StarshipPartFactory.php
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Factory;
+
+use App\Entity\StarshipPart;
+use Zenstruck\Foundry\Persistence\PersistentProxyObjectFactory;
+
+/**
+ * @extends PersistentProxyObjectFactory<StarshipPart>
+ */
+final class StarshipPartFactory extends PersistentProxyObjectFactory
+{
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#factories-as-services
+     *
+     * @todo inject services if required
+     */
+    public function __construct()
+    {
+    }
+
+    public static function class(): string
+    {
+        return StarshipPart::class;
+    }
+
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#model-factories
+     *
+     * @todo add your default values here
+     */
+    protected function defaults(): array|callable
+    {
+        return [
+            'createdAt' => self::faker()->dateTime(),
+            'name' => self::faker()->text(255),
+            'price' => self::faker()->randomNumber(),
+            'updatedAt' => self::faker()->dateTime(),
+        ];
+    }
+
+    /**
+     * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#initialization
+     */
+    protected function initialize(): static
+    {
+        return $this
+            // ->afterInstantiate(function(StarshipPart $starshipPart): void {})
+        ;
+    }
+}
