diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index db4fd60..9112f6b 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -4,6 +4,7 @@ namespace App\DataFixtures;
 
 use App\Entity\StarshipStatusEnum;
 use App\Factory\StarshipFactory;
+use App\Factory\StarshipPartFactory;
 use Doctrine\Bundle\FixturesBundle\Fixture;
 use Doctrine\Persistence\ObjectManager;
 
@@ -36,5 +37,6 @@ class AppFixtures extends Fixture
         ]);
 
         StarshipFactory::createMany(20);
+        StarshipPartFactory::createMany(50);
     }
 }
