diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 4959230..354e723 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -22,21 +22,6 @@ class AppFixtures extends Fixture
             'arrivedAt' => new \DateTimeImmutable('-1 day'),
         ]);
 
-        $starship = new Starship();
-        $starship->setName('USS Taco Tuesday');
-        $starship->setClass('Tex-Mex');
-        $starship->checkIn();
-        $starship->setCaptain('James T. Nacho');
-        $manager->persist($starship);
-
-        $part = new StarshipPart();
-        $part->setName('spoiler');
-        $part->setNotes('There\'s no air drag in space, but it looks cool.');
-        $part->setPrice(500);
-        $manager->persist($part);
-        $part->setStarship($starship);
-        $manager->flush();
-
         StarshipFactory::createOne([
             'name' => 'USS Espresso (NCC-1234-C)',
             'class' => 'Latte',
@@ -54,5 +39,6 @@ class AppFixtures extends Fixture
         ]);
 
         StarshipFactory::createMany(20);
+        StarshipPartFactory::createMany(100);
     }
 }
