diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index a793a77..55ac651 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -39,8 +39,6 @@ class AppFixtures extends Fixture
         ]);
 
         StarshipFactory::createMany(20);
-        StarshipPartFactory::createMany(100, [
-            'starship' => $ship,
-        ]);
+        StarshipPartFactory::createMany(100);
     }
 }
diff --git a/src/Factory/StarshipPartFactory.php b/src/Factory/StarshipPartFactory.php
index 02fc5f5..67d4275 100644
--- a/src/Factory/StarshipPartFactory.php
+++ b/src/Factory/StarshipPartFactory.php
@@ -52,7 +52,7 @@ final class StarshipPartFactory extends PersistentProxyObjectFactory
             'name' => $randomPart[0],
             'notes' => $randomPart[1],
             'price' => self::faker()->randomNumber(5),
-            'starship' => StarshipFactory::new([
+            'starship' => StarshipFactory::randomOrCreate([
                 'status' => StarshipStatusEnum::IN_PROGRESS,
             ]),
         ];
