diff --git a/assets/controllers/closeable_controller.js b/assets/controllers/closeable_controller.js
new file mode 100644
index 0000000..4345d10
--- /dev/null
+++ b/assets/controllers/closeable_controller.js
@@ -0,0 +1,7 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    close() {
+        this.element.remove();
+    }
+}
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 0fea323..f64cf73 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -4,8 +4,14 @@
 
 {% block body %}
     <main class="flex flex-col lg:flex-row flex-1 space-y-4 lg:space-y-0">
-        <aside class="hidden relative w-72 p-2 lg:block lg:h-full bg-gray-800 text-white space-y-4">
-            <button class="absolute top-1 right-1">
+        <aside
+            class="hidden relative w-72 p-2 lg:block lg:h-full bg-gray-800 text-white space-y-4"
+            data-controller="closeable"
+        >
+            <button
+                class="absolute top-1 right-1"
+                data-action="closeable#close"
+            >
                 <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"/><path d="M18 6l-12 12" /><path d="M6 6l12 12" /></svg>
             </button>
 
