diff --git a/templates/main/_shipStatusAside.html.twig b/templates/main/_shipStatusAside.html.twig
index 881f555..aff8de8 100644
--- a/templates/main/_shipStatusAside.html.twig
+++ b/templates/main/_shipStatusAside.html.twig
@@ -1,5 +1,5 @@
 <aside
-    class="pb-8 lg:pb-0 lg:w-[411px] shrink-0 lg:block lg:min-h-screen text-white transition-all overflow-hidden px-8 border-b lg:border-b-0 lg:border-r border-white/20"
+    class="pb-8 lg:pb-0 lg:w-[411px] shrink-0 lg:block lg:min-h-screen text-white transition-all overflow-hidden px-8 border-b lg:border-b-0 lg:border-r border-white/20 transition-all overflow-hidden"
     data-controller="closeable"
 >
     <div class="flex justify-between mt-11 mb-7">
