diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index f64cf73..c8d7d43 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -5,7 +5,7 @@
 {% block body %}
     <main class="flex flex-col lg:flex-row flex-1 space-y-4 lg:space-y-0">
         <aside
-            class="hidden relative w-72 p-2 lg:block lg:h-full bg-gray-800 text-white space-y-4"
+            class="hidden relative w-72 p-2 lg:block lg:h-full bg-gray-800 text-white space-y-4 transition-all overflow-hidden"
             data-controller="closeable"
         >
             <button
