diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 55ac651..ef1da9d 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -2,7 +2,6 @@
 
 namespace App\DataFixtures;
 
-use App\Entity\Starship;
 use App\Entity\StarshipPart;
 use App\Entity\StarshipStatusEnum;
 use App\Factory\StarshipFactory;
diff --git a/templates/starship/show.html.twig b/templates/starship/show.html.twig
index 81e05e5..3c67407 100644
--- a/templates/starship/show.html.twig
+++ b/templates/starship/show.html.twig
@@ -57,7 +57,7 @@
                     <p class="text-[22px] font-semibold">{{ ship.arrivedAt|ago }}</p>
 
                     <h4 class="text-xs text-slate-300 font-semibold mt-2 uppercase">
-                        Parts
+                        Parts ({{ ship.parts|length }})
                     </h4>
                     <ul class="text-sm font-medium text-slate-400 tracking-wide">
                         {% for part in ship.parts %}
