diff --git a/src/Controller/StarshipController.php b/src/Controller/StarshipController.php
index 440af2f..a588e79 100644
--- a/src/Controller/StarshipController.php
+++ b/src/Controller/StarshipController.php
@@ -19,7 +19,6 @@ class StarshipController extends AbstractController
     ): Response {
         return $this->render('starship/show.html.twig', [
             'ship' => $ship,
-            'parts' => $ship->getParts(),
         ]);
     }
 }
diff --git a/templates/starship/show.html.twig b/templates/starship/show.html.twig
index bff4668..81e05e5 100644
--- a/templates/starship/show.html.twig
+++ b/templates/starship/show.html.twig
@@ -60,7 +60,7 @@
                         Parts
                     </h4>
                     <ul class="text-sm font-medium text-slate-400 tracking-wide">
-                        {% for part in parts %}
+                        {% for part in ship.parts %}
                             <li class="border-b border-slate-600 py-2">
                                 <span class="block text-white font-semibold">
                                     {{ part.name }} (✦ {{ part.price }})
