diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 82fe3d6..d27ce14 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -1,6 +1,6 @@
 {% extends 'base.html.twig' %}
 
-{% block title %}{{ parent() }} Starshop: Beam up some parts!{% endblock %}
+{% block title %}Starshop: Beam up some parts!{% endblock %}
 
 {% block body %}
 <h1>
