diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 2176a79..eaad022 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -27,7 +27,7 @@
         </tr>
         <tr>
             <th>Captain</th>
-            <td>{{ myShip.captain|upper }}</td>
+            <td>{{ myShip.captain|upper|lower|title }}</td>
         </tr>
         <tr>
             <th>Status</th>
