diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 18284ac..e34d35f 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -12,9 +12,16 @@ class MainController extends AbstractController
     public function homepage(): Response
     {
         $starshipCount = 457;
+        $myShip = [
+            'name' => 'USS LeafyCruiser (NCC-0001)',
+            'class' => 'Garden',
+            'captain' => 'Jean-Luc Pickles',
+            'status' => 'under construction',
+        ];
 
         return $this->render('main/homepage.html.twig', [
             'numberOfStarships' => $starshipCount,
+            'myShip' => $myShip,
         ]);
     }
 }
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index cf9a7a5..650d725 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -12,3 +12,26 @@
         </p>
     {% endif %}
 </div>
+
+<div>
+    <h2>My Ship</h2>
+
+    <table>
+        <tr>
+            <th>Name</th>
+            <td>{{ myShip.name }}</td>
+        </tr>
+        <tr>
+            <th>Class</th>
+            <td>{{ myShip.class }}</td>
+        </tr>
+        <tr>
+            <th>Captain</th>
+            <td>{{ myShip.captain }}</td>
+        </tr>
+        <tr>
+            <th>Status</th>
+            <td>{{ myShip.status }}</td>
+        </tr>
+    </table>
+</div>
