diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 650d725..2176a79 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -27,7 +27,7 @@
         </tr>
         <tr>
             <th>Captain</th>
-            <td>{{ myShip.captain }}</td>
+            <td>{{ myShip.captain|upper }}</td>
         </tr>
         <tr>
             <th>Status</th>
