diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 171ab06..ff5a71c 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -11,6 +11,6 @@ class MainController extends AbstractController
     #[Route('/')]
     public function homepage(): Response
     {
-        return new Response('<strong>Starshop</strong>: your monopoly-busting option for Starship parts!');
+        return $this->render('main/homepage.html.twig');
     }
 }
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
new file mode 100644
index 0000000..89c82d7
--- /dev/null
+++ b/templates/main/homepage.html.twig
@@ -0,0 +1,3 @@
+<h1>
+    Starshop: your monopoly-busting option for Starship parts!
+</h1>
