diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index e0de37f..f39c6a4 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -1,5 +1,6 @@
 {% extends 'base.html.twig' %}
 
+{% block body %}
 <h1>
     Starshop: your monopoly-busting option for Starship parts!
 </h1>
@@ -37,3 +38,4 @@
         </tr>
     </table>
 </div>
+{% endblock %}
