diff --git a/templates/starship/show.html.twig b/templates/starship/show.html.twig
index 619e944..d767faf 100644
--- a/templates/starship/show.html.twig
+++ b/templates/starship/show.html.twig
@@ -32,6 +32,9 @@
 
                 <h4 class="text-xs text-slate-300 font-semibold mt-2 uppercase">Ship Status</h4>
                 <p class="text-[22px] font-semibold">30,000 lys to next service</p>
+
+                <h4 class="text-xs text-slate-300 font-semibold mt-2 uppercase">Arrived At</h4>
+                <p class="text-[22px] font-semibold">{{ ship.arrivedAt|ago }}</p>
             </div>
         </div>
     </div>
